Lib/test/test_pyrepl/test_pyrepl.py

```diff
@@ -1123,3 +1123,26 @@ def run_repl(
             process.kill()
             exit_code = process.wait()
         return "\n".join(output), exit_code
+
+    def test_readline_history_file(self):
+        # skip, if readline module is not available
+        readline = import_module('readline')
+        if readline.backend != "editline":
+            self.skipTest("GNU readline is not affected by this issue")
```

Comment:
Since what we're trying to do is to fix reading *old* history files, you could add three history files to test data:
* an editline history file
* a readline history file
* a pyrepl history file with multiline blocks

That way all three would have tests and we wouldn't regress on all three fronts.

Reply:
This will have to wait for a future enhancement.